llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java

```diff
@@ -147,7 +152,9 @@ public void notifyLock(LlapCacheableBuffer buffer) {
 
   @Override
   public void notifyUnlock(LlapCacheableBuffer buffer) {
-
+    if (proactiveEvictionEnabled) {
```

Comment:
Why is this?

Reply:
notifyUnlock(buffer) is invoked whenever a buffer was accessed (and the read from it has been finished)
This here covers a very rare scenario, details in the test here: https://github.com/apache/hive/pull/1675/files#diff-826be47ae17dbe2de96b3a40cf13fcc40604c5f630fd95d2732a646318ef33d4R307-R311

Reply:
Maybe add a comment, if we have to push any changes?

Reply:
ok